library/src/main/java/com/otaliastudios/zoom/internal/movement/PanManager.kt

```diff
@@ -144,16 +153,18 @@ internal class PanManager(provider: () -> MatrixController) : MovementManager(pr
     }
 
     /**
-     * The amount of overscroll that is allowed in both direction. This is currently
-     * a fixed value, but might be made configurable in the future.
+     * The amount of overscroll that is allowed in horizontal direction.
      */
     @ZoomApi.ScaledPan
-    internal val maxOverPan: Float
-        get() {
-            val overX = controller.containerWidth * DEFAULT_OVERPAN_FACTOR
-            val overY = controller.containerHeight * DEFAULT_OVERPAN_FACTOR
-            return min(overX, overY)
-        }
+    internal val maxHorizontalOverPan: Float
+        get() = overPanRangeProvider.getHorizontalOverPanRange(engine)
```

Comment:
Same here, I think we should abs() the user value. Or max(0, value) ?

Reply:
I like max(0, value) better. I will also add a warning log.